rotkehlchen/api/rest.py

```diff
@@ -4140,3 +4140,46 @@ def add_evm_transaction_by_hash(
             )
         evm_manager.transactions_decoder.decode_transaction(transaction=transaction, tx_receipt=tx_receipt)  # noqa: E501
         return OK_RESULT
+
+    @async_api_call()
+    def get_binance_savings_balance(
+            self,
+            location: Literal[Location.BINANCE, Location.BINANCEUS],
+    ) -> dict[str, Any]:
+        """
+        Returns savings balance alongside total interests accrued in the
+        specified exchange location i.e. Binance/BinanceUS
+        """
+        exchanges_list = self.rotkehlchen.exchange_manager.connected_exchanges.get(location)
+        if exchanges_list is None:
+            return wrap_in_fail_result(
+                message=f'There is no {location.name} account connected.',
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        balances: dict[str, defaultdict[AssetWithOracles, Balance]] = {
+            'savings_balance': defaultdict(Balance),
+            'interest_gained': defaultdict(Balance),
```

Comment:
why does this need to be a defaultdict? Can't it be

```suggestion
            'savings_balance': {},
            'interest_gained': {},
```

?